src/golang/lib/workflow/artifact/response.go

```diff
@@ -23,9 +23,17 @@ type Response struct {
 
 type RawResultResponse struct {
 	// Contains only the `result`. It mostly mirrors 'artifact_result' schema.
-	Id          uuid.UUID              `json:"id"`
-	ContentPath string                 `json:"content_path"`
-	ExecState   *shared.ExecutionState `json:"exec_state"`
+	Id uuid.UUID `json:"id"`
+
+	// If `ContentSerialized` is set, the content is small and we directly send
+	// it as a part of response. It's consistent with the object stored in `ContentPath`.
+	//
+	// Otherwise, the content is large and
+	// one should send an additional request to fetch the content.
+	ContentPath       string  `json:"content_path"`
+	ContentSerialized *string `json:"content_serialized"`
```

Comment:
this is intentionally put as as pointer to distinguish 'value not available' vs 'value being an empty string'

Reply:
Can you note what it means for the content to be "serialized"? It's the raw bytes data at the storage path, converted to a string right?